cocotb/utils.py

```diff
@@ -578,3 +579,36 @@ def remove_traceback_frames(tb_or_exc, frame_names):
             assert tb.tb_frame.f_code.co_name == frame_name
             tb = tb.tb_next
         return tb
+
+
+def walk_coro_stack(coro):
+    """Walk down the coroutine stack. Supports coroutines and generators."""
+    while coro is not None:
+        try:
+            f = getattr(coro, 'cr_frame')
+            coro = coro.cr_await
+        except AttributeError:
+            try:
+                f = getattr(coro, 'gi_frame')
+                coro = coro.gi_yieldfrom
+            except AttributeError:
+                f = None
+                coro = None
+        if f is not None:
+            yield (f, f.f_lineno)
+
+
+def extract_coro_stack(coro, limit=None):
```

Comment:
This is based on `traceback.extract_tb`, using the `StackSummary.extract` classmethod for ease of use.
I think it might be possible to get the `__qualname__` for each coroutine in the stack and display that.